kore/src/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -123,16 +125,19 @@ termEquals
     axiomIdToSimplifier
     first
     second
-  = do
-    result <-
-        termEqualsAnd
+  = MaybeT $ do
```

Comment:
One thing we could do to make this a little less awkward is remove the `MaybeT` and use `empty` in the unifier's `BranchT` to represent "not handled". We would have to return an explicit `bottom` when a case is handled and the unifier is `\bottom`, but I think that would also avoid the problems with explanations in the REPL.